compiler/rustc_trait_selection/src/solve/project_goals.rs

```diff
@@ -131,8 +131,14 @@ impl<'tcx> assembly::GoalKind<'tcx> for ProjectionPredicate<'tcx> {
             else {
                 return
             };
-
-            let nested_goals = obligations.into_iter().map(|o| o.into()).collect();
+            let where_clause_bounds = tcx
+                .predicates_of(impl_def_id)
+                .instantiate_own(tcx, impl_substs)
```

Comment:
while this shouldn't matter as impls don't have a parent, I still think we should use `instantiate` here. It may safe us if we're ever adding module generics or something like that.